### PR TITLE
Improved pan extension

### PR DESCRIPTION
**Pull request by Fueredoriku**
- Pan now blocks explosion damage if aimed correctly
panSplosion
- Pan skating speed increased a lot
- Add pan-shot! popup text when blocking damage

panShot

### CODE REVIEW

**Review by toberge — CHANGES_REQUESTED** [review]
Nice that it finally blocks explosions 🎉 (though sometimes not quite)

However, currently it seems that bullets are blocked by the huge explosion catchers as well, which makes it impossible to hit someone from the front 🙃 
image

File: Assets/Scripts/Augment/AugmentImplementations/Pan.cs [hunk 1]
```diff
@@ -22,30 +26,48 @@ void Start()
...
+            .ToList().ForEach(box =>  box.health = health);
+        if (playerMovement)
+            playerMovement.OnMove += TryPanSkateBoost;
     }
 
     private void TryTrickJump(InputAction.CallbackContext ctx)
     {
         if (playerMovement)
-            if (playerMovement.StateIsAir)
+            if (playerMovement.StateIsAir && playerMovement.IsCrouching)
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/Pan.cs]: at this point the `if` pyramid looks a little odd

File: Assets/Scripts/Augment/AugmentImplementations/PanHitbox.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PanHitbox : HitboxController
+{
+    [SerializeField]
+    private HealthController panHealth;
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/PanHitbox.cs:8]: Superfluous. The superclass already has a public health member.

Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/PanHitbox.cs:8]: They have two different purposes, the panhealth is for the panshotemitter, not for tracking actual damage!

File: Assets/Scripts/Augment/AugmentImplementations/Pan.cs (hunk 1 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/Pan.cs]: Hehehe, I concede that the pyramid should go, I will miss it tho 🫡

File: Assets/Scripts/Augment/AugmentImplementations/PanHitbox.cs (hunk 2 above)
Comment (toberge) [reply Assets/Scripts/Augment/AugmentImplementations/PanHitbox.cs:8]: Oh okay

**Review by toberge — APPROVED** [review]
Works great now!

**Pull request merged**